openstack/03a-IPHC/iphc.c

```diff
@@ -560,147 +562,146 @@ void iphc_retrieveIPv6Header(OpenQueueEntry_t* msg, ipv6_header_iht* ipv6_outer_
-              ipinip_length = temp_8b & IPINIP_LEN_6LORH_MASK;
-              ipv6_outer_header->header_length += 1;
-              temp_8b = *(uint8_t*)(msg->payload+ipv6_outer_header->header_length+*page_length+extention_header_length);
-              if (temp_8b == IPINIP_TYPE_6LORH){
-                  // this is IpinIP 6LoRH
-                  ipv6_outer_header->header_length += 1;
-                  ipv6_outer_header->hop_limit = *(uint8_t*)(msg->payload+ipv6_outer_header->header_length+*page_length+extention_header_length);
-                  ipv6_outer_header->header_length += 1;
-                  // destination address maybe is the first address in RH3 6LoRH OR dest adress in IPHC, reset first
-                  // update destination address if necessary after the processing
-                  ipv6_outer_header->dest.type = ADDR_NONE;
-                  memset(&(ipv6_outer_header->dest.addr_128b[0]),0,16);
-                  if (ipinip_length == 1){
-                      // source address is root
-                      memset(&(ipv6_outer_header->src),0,sizeof(open_addr_t));
-                      packetfunctions_mac64bToIp128b(idmanager_getMyID(ADDR_PREFIX),(open_addr_t*)dagroot_mac64b,&(ipv6_outer_header->src));
-                      // destination address is the first address in RH3 6LoRH OR dest adress in IPHC
-                  } else{
-                      switch(ipinip_length-1){
-                      case 16:
-                           // this is source address 
-                           packetfunctions_readAddress(((uint8_t*)(msg->payload+ipv6_outer_header->header_length+*page_length+extention_header_length)),ADDR_128B,&ipv6_outer_header->src,OW_BIG_ENDIAN);
-                           ipv6_outer_header->header_length += 16*sizeof(uint8_t);
-                           break;
-                      case 8:
-                           // this is source address 
-                           packetfunctions_readAddress(((uint8_t*)(msg->payload+ipv6_outer_header->header_length+*page_length+extention_header_length)),ADDR_64B,&temp_addr_64b,OW_BIG_ENDIAN);
-                           ipv6_outer_header->header_length += 8*sizeof(uint8_t);
-                           packetfunctions_mac64bToIp128b(idmanager_getMyID(ADDR_PREFIX),&temp_addr_64b,&ipv6_outer_header->src);
-                           break;
-                      default:
-                           // do not support other length yet and destination address will be in RH3 or IPHC
-                           openserial_printError(
-                              COMPONENT_IPHC,
-                              ERR_6LOWPAN_UNSUPPORTED,
-                              (errorparameter_t)12,
-                              (errorparameter_t)(ipinip_length-1)
-                           );
-                      }
-                  }
-              } else {
-                  // don't defined yet
-                  openserial_printError(

```

Comment:
Hi @zach-b , the fix is good. For the seialprint, I would like to keep it It's good to be aware when receiving unsupported routing header. :)